lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1244,6 +1214,20 @@ public function prepare_item_for_response( $post, $request ) {
 	}
 
 	/**
+	 * Overwrite the default protected title format.
+	 *
+	 * By default WordPress will show password protected posts with a title of
+	 * "Protected: %s", as the REST API communicates the protected status of a post
+	 * in a machine readable format, we remove the "Protected: " prefix.
```

Comment:
Minor quibble: including "Protected: " has nothing to do with machine readability and everything to do with that string being purely presentational. I'd alter the explanation of this decision here.